File: code/dobot_quad_sdk-main/high_level/python/examples/e1_get_available_motions.py
#!/usr/bin/env python3
"""Example 1: List all available motions and their parameters."""
import sys
from dobot_quad import RobotClient


def main():
    robot = RobotClient(sys.argv[1] if len(sys.argv) > 1 else "192.168.5.2:50051")
    res = robot.get_motions()

    if not res.success:
        print(f"Failed: {res.message}")
        return 1

    print(f"Available motions ({len(res.motions)}):\n")
    for m in res.motions:
        desc = res.descriptions.get(m.motion_id, "")
        print(f"  [{m.motion_id}] {desc}")
        for p in m.parameters:
            val = (
                p.float_value if p.HasField("float_value") else p.int_value
                if p.HasField("int_value") else f'"{p.string_value}"' if p.HasField("string_value")
                else p.bool_value if p.HasField("bool_value") else "(unset)"
            )
            print(f"      {p.key}: {val}")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: code/dobot_quad_sdk-main/high_level/cpp/e1_get_available_motions.cpp
// Example 1: List all available motions and their parameters.
// Usage: ./e1_get_available_motions [server_address]

#include "robot_client.h"

int main(int argc, char** argv)
{
    robot::Client client(argc > 1 ? argv[1] : "192.168.5.2:50051");
    auto res = client.get_motions();

    if (!res.success()) {
        std::cerr << "Failed: " << res.message() << std::endl;
        return 1;
    }

    std::cout << "Available motions (" << res.motions_size() << "):\n" << std::endl;

    for (const auto& m : res.motions()) {
        std::cout << "  [" << m.motion_id() << "]";
        auto it = res.descriptions().find(m.motion_id());
        if (it != res.descriptions().end())
            std::cout << " - " << it->second;
        std::cout << std::endl;

        for (const auto& p : m.parameters()) {
            std::cout << "      " << p.key() << ": ";
            switch (p.value_case()) {
                case grpc_comm::Parameter::kFloatValue:
                    std::cout << p.float_value();
                    break;
                case grpc_comm::Parameter::kIntValue:
                    std::cout << p.int_value();
                    break;
                case grpc_comm::Parameter::kStringValue:
                    std::cout << '"' << p.string_value() << '"';
                    break;
                case grpc_comm::Parameter::kBoolValue:
                    std::cout << (p.bool_value() ? "true" : "false");
                    break;
                default:
                    std::cout << "(unset)";
            }
            std::cout << std::endl;
        }
    }
    return 0;
}
